samza-core/src/main/java/org/apache/samza/operators/functions/PartialJoinFunction.java

```diff
@@ -47,23 +48,6 @@
    *
    * @return the key value store containing the state for this stream
    */
-  KeyValueStore<K, PartialJoinMessage<M>> getState();
+  KeyValueStore<K, TimestampedValue<M>> getState();
```

Comment:
Should this be 
`KVStore<TimeSeriesKey<K>, V>`? Why do we need to store the timestamp as a part of the value?  
You can also perhaps use the same operator for doing windowed-joins later?

Reply:
Because we need to look up by key for doing joins b/w the two streams, and timestamps are not part of the key. I don't want to use a timeline store for this version since all queries would be range queries which is not caching-friendly. We can refactor if necessary when we do other kind of joins.